docs/src/main/sphinx/connector/snowflake.rst

```diff
@@ -0,0 +1,91 @@
+===================
+Snowflake connector
+===================
+
+The Snowflake connector allows querying and creating tables in an
+external Snowflake account. This can be used to join data between
+different systems like Snowflake and Hive, or between two different
+Snowflake accounts.
+
+Configuration
+-------------
+
+To configure the Snowflake connector, create a catalog properties file
+in ``etc/catalog`` named, for example, ``snowflake.properties``, to
+mount the Snowflake connector as the ``snowflake`` catalog.
+Create the file with the following contents, replacing the
+connection properties as appropriate for your setup:
+
+.. code-block:: none
+
+    connector.name=snowflake
+    connection-url=jdbc:snowflake://<account>.snowflakecomputing.com
+    connection-user=root
+    connection-password=secret
+    snowflake.account=account
+    snowflake.database=database
+    snowflake.role=role
+    snowflake.warehouse=warehouse
+
+
+Multiple Snowflake Databases or Accounts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Snowflake connector can only access a single database within
+a Snowflake account. Thus, if you have multiple Snowflake databases,
+or want to connect to multiple Snowflake accounts, you must configure
+multiple instances of the Snowflake connector.
+
+.. snowflake-type-mapping:
+Type mapping
+------------
+
+Trino supports the following Snowflake data types:
+
+==================================  ===============================
+Snowflake Type                      Trino Type
+==================================  ===============================
+``boolean``                         ``boolean``
+``tinyint``                         ``bigint``
+``smallint``                        ``bigint``
+``int``                             ``bigint``
+``integer``                         ``bigint``
+``bigint``                          ``bigint``
+``float``                           ``real``
+``real``                            ``real``
+``double``                          ``double``
+``varchar(n)``                      ``varchar(n)``
+``char(n)``                         ``char(n)``
+``text``                            ``varchar(16777216)``
+``date``                            ``date``
+``time(n)``                         ``time(n)``
+==================================  ===============================
+
+Complete list of `Snowflake data types
+<https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html>`_.
+
+.. _snowflake-pushdown:
+
+Pushdown
+--------
+Query pushdown is supported with v2.1 (and later) of the Snowflake Connector for Spark.
```

Comment:
I'm sorry for that, In fact I'm not sure if snowflake supports the pushdown because I didn't find any info related except an article about the snowflake using in spark. I'll remove this comment. 😄